getsolus/packages
# grub2: Update to 2.12

**Summary**

New in 2.12:

* GCC 13 support.
* clang 14 support.
* binutils 2.38 support.
* Unification of EFI Linux kernel loader across architectures.
* Transition to EFI Linux kernel stub loader for x86 architecture.
* Initial support for Boot Loader Interface.
* Support for dynamic GRUB runtime memory addition using firmware calls.
* PCI and MMIO UARTs support.
* SDL2 support.
* LoongArch support.
* TPM driver fixes.
* Many filesystems fixes.
* Many CVE and Coverity fixes.
* Debugging support improvements.
* Tests improvements.
* Documentation improvements.
* ...and tons of other fixes and cleanups...

New in 2.06:

* GCC 10 support.
* clang 10 support.
* SBAT support.
* LUKS2 support.
* Drop small MBR gap support.
* Xen Security Modules (XSM/FLASK) support.
* The lockdown mechanism similar to the Linux kernel one.
* Disable the os-prober by default.
* Many backports of GRUB distros specific patches.
* BootHole and BootHole2 fixes.
* ...and tons of other fixes and cleanups...

Resolves #1348

**Security**
Includes fixes for:
- CVE-2014-4607
- CVE-2020-10713
- CVE-2020-14308
- CVE-2020-14309
- CVE-2020-14310
- CVE-2020-14311
- CVE-2020-14372
- CVE-2020-15705
- CVE-2020-15706
- CVE-2020-15707
- CVE-2020-25632
- CVE-2020-25647
- CVE-2020-27749
- CVE-2020-27779
- CVE-2021-20225
- CVE-2021-20233
- CVE-2021-3418
- CVE-2021-3695
- CVE-2021-3696
- CVE-2021-3697
- CVE-2021-3981
- CVE-2022-2601
- CVE-2022-28733
- CVE-2022-28734
- CVE-2022-28735
- CVE-2022-28736
- CVE-2022-3775
- CVE-2023-4692
- CVE-2023-4693

**Test Plan**

- Install Solus with various configurations:
  ![Screenshot 2024-05-01 at 14-11-39 Screenshot](https://github.com/getsolus/packages/assets/5798032/db327c57-81f1-4323-a69a-4641ea34257a)

- Update VMs in plain, LUKS and LVM form and verify reboot.

**Checklist**
